applications/OptimizationApplication/python_scripts/controls/shape/vertex_morphing_shape_control.py

```diff
@@ -181,93 +54,66 @@ def __init__(self, name: str, model: Kratos.Model, parameters: Kratos.Parameters
-        else:
-            self.filter_model_part = self.model_part_operation.GetModelPart()
+        self.model_part = self.model_part_operation.GetModelPart()
 
         # initialize filters
-        if not self.is_filter_implicit:
-            # make sure that explicit filter model part only has conditions
-            if len(self.filter_model_part.Elements) > 0:
-                raise RuntimeError("VertexMorphingShapeControl with explicit filtering only allows model parts with conditions")
-            # now damping and creating the filter
-            fixed_model_parts_names = list(self.parameters["fixed_model_parts"].keys())
-            if len(fixed_model_parts_names)>0:
-                self.fixed_model_part_operation = ModelPartOperation(self.model, ModelPartOperation.OperationType.UNION, f"control_{self.GetName()}", fixed_model_parts_names, False)
-                self.fixed_model_part = self.fixed_model_part_operation.GetModelPart()
-                self.filter = KratosOA.NodalExplicitFilter(self.filter_model_part, self.fixed_model_part, self.parameters["filter_settings"]["filter_function_type"].GetString(),
-                                                           self.parameters["filter_settings"]["damping_function_type"].GetString(),
-                                                           self.parameters["filter_settings"]["max_nodes_in_filter_radius"].GetInt())
-            else:
-                self.filter = KratosOA.NodalExplicitFilter(self.filter_model_part, self.parameters["filter_settings"]["filter_function_type"].GetString(),
-                                                           self.parameters["filter_settings"]["max_nodes_in_filter_radius"].GetInt())
-
-            filter_radius_field = Kratos.Expression.NodalExpression(self.filter_model_part)
-            Kratos.Expression.LiteralExpressionIO.SetData(filter_radius_field, self.parameters["filter_settings"]["radius"].GetDouble())
-            self.filter.SetFilterRadius(filter_radius_field)
-            self.control_field = Kratos.Expression.NodalExpression(self.filter_model_part)
-            Kratos.Expression.LiteralExpressionIO.SetData(self.control_field,[0,0,0])
-        else:
-            self.filter.Initialize()
-            self._SetFixedModelPartValues(False)
-            self.control_field = self.filter.UnFilterField(self.GetPhysicalField())
-            self._SetFixedModelPartValues(True)
+        self.filter.SetComponentDataView(ComponentDataView(self, self.optimization_problem))
+        self.filter.Initialize()
+
+        # getting UnfilteredField of the PhysicalField is not possible with
+        # explicit solvers, and is not useful for this control hence, this
+        # will only update the shape changes to the existing shape. Hence,
+        # taking the actual physical field as the control field initially.
+        self.control_field = self.GetPhysicalField()
```

Comment:
that's not true ! Please do not change the original implementation without providing notices and reasons. Please take the original implementation which distinguishes between explicit and implicit filters. Implicit filter provides the control field, hence filters should provide unfilterfield, in case of explicit, zerofield should be returned

Reply:
Please complete the comments with the fact that implicit filters provide the control field by unfiltering however explicit filters only return zero field since thy can not !

Reply:
In overall, please only use the unfilterfield function of the filter to get the control field

Reply:
Also do not use physical field as control field since it is very confusing and use the comments I made above that if the filter is explicit unfilter field returns zero otherwise implicit filter returns the control field

Reply:
understood, and will be changed